File: src/codes/Users_Manual/Tutorial/silencer/disable.py
~import numpy as np
~from pyautd3 import AUTD3, Controller, Hz, FociSTM, Silencer
~from pyautd3.link.nop import Nop
~autd = Controller.open([AUTD3(pos=[0.0, 0.0, 0.0], rot=[1, 0, 0, 0])], Nop())
autd.send(Silencer.disable())

center = autd.center() + np.array([0.0, 0.0, 150.0])
point_num = 20
radius = 30.0
autd.send(
    FociSTM(
        foci=(
            center + radius * np.array([np.cos(theta), np.sin(theta), 0])
            for theta in (2.0 * np.pi * i / point_num for i in range(point_num))
        ),
        config=50.0 * Hz,
    )
)

File: src/codes/Users_Manual/Tutorial/silencer/disable.rs
# use autd3::prelude::*;
# fn main() -> Result<(), Box<dyn std::error::Error>> {
# let mut autd = Controller::open([AUTD3::default()], Nop::new())?;
autd.send(Silencer::disable())?;

let center = autd.center() + Vector3::new(0.0, 0.0, 150. * mm);
let point_num = 20;
let radius = 30.0 * mm;
autd.send(FociSTM {
    foci: (0..point_num)
        .map(|i| {
            let theta = 2.0 * PI * i as f32 / point_num as f32;
            let p = radius * Vector3::new(theta.cos(), theta.sin(), 0.0);
            center + p
        })
        .collect::<Vec<_>>(),
    config: 50.0 * Hz,
})?;
# Ok(())
# }